lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3997,6 +3997,7 @@ void ByteCodeGenerator::StartEmitFunction(ParseNode *pnodeFnc)
                 !funcInfo->Escapes() &&
                 funcInfo->frameObjRegister != Js::Constants::NoRegister &&
                 !ApplyEnclosesArgs(pnodeFnc, this) &&
+                !pnodeFnc->sxFnc.HasDefaultArguments() &&
```

Comment:
Is it ok to do cached scopes in other cases of non-simple params - rest, destructuring patterns?